dali/python/nvidia/dali/pipeline.py

```diff
@@ -209,22 +206,79 @@ def max_batch_size(self):
 
     @property
     def num_threads(self):
-        """Number of CPU threads used by the pipeline."""
+        """Number of CPU threads used by this pipeline."""
         return self._num_threads
 
     @property
     def device_id(self):
-        """Id of the GPU used by the pipeline."""
-        return self._device_id
+        """Id of the GPU used by the pipeline or None for CPU-only pipelines."""
+        return None if self._device_id == types.CPU_ONLY_DEVICE_ID else self._device_id
+
+    @property
+    def seed(self):
+        """Random seed used in the pipeline or None, if seed is not fixed."""
```

Comment:
You are not checking if `seed` is not fixed(?)

Reply:
I removed replacement of None with -1 in the constructor.